Source/Scene/AutoExposure.js

```diff
@@ -0,0 +1,416 @@
+        }
+
+        source +=
+            '    gl_FragColor = vec4(color); \n' +
+            '} \n';
+        return source;
+    }
+
+    function createCommands(autoexposure, context) {
+        destroyCommands(autoexposure);
+        var framebuffers = autoexposure._framebuffers;
+        var length = framebuffers.length;
+
+        var commands = new Array(length);
+
+        for (var i = 0; i < length; ++i) {
+            commands[i] = context.createViewportQuadCommand(getShaderSource(i, length), {
+                framebuffer : framebuffers[i],
+                uniformMap : createUniformMap(autoexposure, i)
+            });
+        }
+        autoexposure._commands = commands;
+    }
+
+    /**
+     * A function that will be called before execute. Used to clear any textures attached to framebuffers.
+     * @param {Context} context The context.
+     * @private
+     */
+    AutoExposure.prototype.clear = function(context) {
+        var framebuffers = this._framebuffers;
+        if (!defined(framebuffers)) {
+            return;
+        }
+
+        var clearCommand = this._clearCommand;
+        if (!defined(clearCommand)) {
+            clearCommand = this._clearCommand = new ClearCommand({
+                color : new Color(0.0, 0.0, 0.0, 0.0),
+                framebuffer : undefined
+            });
+        }
+
+        var length = framebuffers.length;
+        for (var i = 0; i < length; ++i) {
+            clearCommand.framebuffer = framebuffers[i];
+            clearCommand.execute(context);
+        }
+    };
+
+    /**
+     * A function that will be called before execute. Used to create WebGL resources and load any textures.
+     * @param {Context} context The context.
+     * @private
+     */
+    AutoExposure.prototype.update = function(context) {
+        var width = context.drawingBufferWidth;
+        var height = context.drawingBufferHeight;
+
+        if (width !== this._width || height !== this._height) {
+            this._width = width;
+            this._height = height;
+
+            createFramebuffers(this, context);
+            createCommands(this, context);
+
+            if (!this._ready) {
+                this._ready = true;
+            }
+        }
+
+        var framebuffers = this._framebuffers;
+        var temp = framebuffers[framebuffers.length - 1];
+        framebuffers[framebuffers.length - 1] = this._previousLuminance;
+        this._commands[this._commands.length - 1].framebuffer = this._previousLuminance;
+        this._previousLuminance = temp;
+    };
+
+    /**
+     * Executes the post-process stage. The color texture is the texture rendered to by the scene or from the previous stage.
+     * @param {Context} context The context.
+     * @param {Texture} colorTexture The input color texture.
+     * @private
+     */
+    AutoExposure.prototype.execute = function(context, colorTexture) {
+        this._colorTexture = colorTexture;
+
+        var commands = this._commands;
```

Comment:
Do other post-processing stages or primitives have code like this to execute an array of commands?  If so, can we move it to a reusable function?